cmd/solution/zap.go

```diff
@@ -0,0 +1,164 @@
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package solution
+
+import (
+	"fmt"
+	"net/url"
+	"os"
+	"path/filepath"
+	"reflect"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/spf13/cobra"
+
+	"github.com/cisco-open/fsoc/config"
+	"github.com/cisco-open/fsoc/output"
+)
+
+var solutionZapCmd = &cobra.Command{
+	Use:   "zap <solution-name>",
+	Args:  cobra.MaximumNArgs(1),
+	Short: "Upload an empty version of a solution to clean it up",
+	Long: `This command creates an empty version of an existing solution and uploads it.
+
+This is for the purpose of cleaning up a solution by removing the knowledge types and knowledge objects
+associated with it.  Use this command with caution.`,
+	Example:          `  fsoc solution zap mysolution`,
+	Run:              zapSolution,
+	Annotations:      map[string]string{config.AnnotationForConfigBypass: ""},
+	TraverseChildren: true,
+}
+
+func getSolutionZapCmd() *cobra.Command {
+
+	solutionZapCmd.Flags().
+		String("tag", "", "Tag associated with the solution to zap (required)")
+
+	_ = solutionZapCmd.MarkFlagRequired("tag")
+
+	solutionZapCmd.Flags().
+		Int("wait", -1, "Wait to terminate the command until the solution is install succesfully")
+
+	solutionZapCmd.Flags().
+		Bool("yes", false, "Skip warning message and bypass confirmation step")
+
+	return solutionZapCmd
+}
+
+func zapSolution(cmd *cobra.Command, args []string) {
+	var confirmationAnswer string
+	var solutionPathInFlag string
+	var solutionId string
+	var solutionInstallObjectQuery string
+	var lastSolutionInstallVersion string
+	var solutionName string
+
+	solutionInstallObjectChan := make(chan StatusItem)
+	solutionObjectChan := make(chan ExtensibilitySolutionObjectData)
+
+	cfg := config.GetCurrentContext()
+	solutionTag, _ := cmd.Flags().GetString("tag")
+	skipConfirmationMessage, _ := cmd.Flags().GetBool("yes")
+	cmd.Flags().StringVar(&solutionName, "solution-name", "", "")
+	cmd.Flags().StringVar(&solutionPathInFlag, "solution-bundle", "", "")
+	cmd.Flags().StringVar(&lastSolutionInstallVersion, "solution-version", "", "")
+
+	solutionName = getSolutionNameFromArgs(cmd, args, "")
+	solutionName = strings.ToLower(solutionName)
+
+	if !skipConfirmationMessage {
+		fmt.Printf("WARNING! This command will remove all of the objects and types that are associated with this solution and will purge all data related to those objects and types.  Proceed with caution!  \nPlease type YES and hit enter confirm that you want to zap the solution with name: %s and tag: %s \n", solutionName, solutionTag)
+		fmt.Scanln(&confirmationAnswer)
+
+		if confirmationAnswer != "YES" {
```

Comment:
instead of yes, can we ask for the solution name? Either just the name or the name and the tag